src/client/pydaos/pydaos_shim.c

```diff
@@ -378,24 +441,39 @@ static PyObject *
 __shim_handle__obj_idgen(PyObject *self, PyObject *args)
 {
 	PyObject		*return_list;
-	daos_handle_t		 coh;
+	struct open_handle	*hdl;
 	daos_oclass_id_t	 cid;
 	int			 cid_in;
 	daos_obj_id_t		 oid;
+	int			 rc = DER_SUCCESS;
 
 	/* Parse arguments */
-	RETURN_NULL_IF_FAILED_TO_PARSE(args, "Li", &coh.cookie, &cid_in);
+	RETURN_NULL_IF_FAILED_TO_PARSE(args, "Ki", &hdl, &cid_in);
 	cid = (uint16_t) cid_in;
 
-	/** XXX: OID should be generated via daos_cont_alloc_oids() */
-	srand(time(0));
-	oid.lo = rand();
-	oid.hi = 0;
+	/** If we ran out of local OIDs, alloc one from the container */
+	if (hdl->alloc.hi >= MAX_OID_HI) {
+		rc = daos_cont_alloc_oids(hdl->coh, 1, &hdl->alloc.lo, NULL);
+		if (rc) {
+			D_ERROR("daos_cont_alloc_oids() Failed (%d)\n", rc);
```

Comment:
Formatted rc?
```suggestion
			D_ERROR("daos_cont_alloc_oids() Failed "DF_RC"\n", DP_RC(rc));
```

Reply:
fair enough. I will fix in my next patch.